### ISSUE TITLE
[feature] ability to auto backup pictures and to remove pictures from local device

### ISSUE THREAD

**Issue by spupuz** [alextran1502/immich#165]
could you please add ability to auto backup pictures and to remove pictures from local device (android phone) when backup is done?

**Comment by Tschuuuls** [comment]
It would probably be useful to set a threshold on the age of photos. Like delete everything that's backed up and older than 2 Years for example.

**Comment by bertmelis** [comment]
Actually, just like Google Photos I'd like to delete photos from the backup (server) as well. When background backup is implemented, all photos will be transferred, also the ones that you delete from your phone. (Who doesn't take multiple pictures only to save the best one?)

**Comment by zackpollard** [comment]
Yea, would be good to have both all options available to you, could just say Delete From: Server, Local, Both as clickable options

**Comment by r3nor** [comment]
This feature would really be a game-changer for Immich. It is the one feature I've been missing from all Google Photos replacements!

**Comment by mhqschen** [comment]
https://github.com/immich-app/immich/pull/774 is closed? Is this feature still under development?

**Comment by spupuz** [comment]
Any news about that?

**Comment by bo0tzz** [comment]
That PR was closed because it was inactive for months, so the feature is not currently under development.

**Comment by spupuz** [comment]
well :/ not good once people backup photos to a "cloud" or self-hosted cloud would probably regain the space on the device that was the original source

**Comment by alextran1502** [comment]
This feature will be implemented eventually, however it is not a priority at the moment

**Comment by acjohnson** [comment]
I decided to give Immich a go on kubernetes with the helm chart and I gotta say this is really well done so great work 😃

(Facial recognition was the feature that pushed me to give it a try btw.)

This "Free up space" ability (as Google/Android calls it) is exactly the missing feature that is needed to eventually completely drop Google Photos IMHO.

Again great work I'm amazed and encouraged 🎉

**Comment by Sniper435** [comment]
This is a feature I'd also love to see, I've just set up an immich server to determine whether it's a suitable, self hosted, Google photos replacement and so far this is the biggest missing feature I've come across.

**Comment by nmoore1** [comment]
I'm surprised this isn't already a feature. I'm blown away that it isn't a priority feature. 

I can't speak for everyone, but this feature is the sole reason I spun up an instance of immich. To give my family an easy way to offload photos to my server and free up device space following. 

PLEASE put this in the feature cycle to be worked on. 

Otherwise, you guys have done a phenominal job on all fronts. Functionality is great, ui is smooth and the mobile apps at this stage of development is mind blowing. 

Thank you!

**Comment by alextran1502** [comment]
@nmoore1 we are aware this is a needed feature and will work on it eventually. It is one of the last feature we will work on before the stable release. Because of the nature of destruction of the feature, we want to make sure the server is stable and you have a good way to rebuild your instance if needed during the development cycle.

**Comment by nmoore1** [comment]
@alextran1502 that's great to hear! I was worried it wouldn't be a thing at all, but that makes sense. Thank you. 

Also, looks like deleting from the native photo app accomplishes this in a round about way. I set up a one way back up to make sure nothing goes sideways while still in the development phase.

**Comment by Srylax** [comment]
@alextran1502 Thanks for your answer. Is there any tracking issue for this stable release or a planned release date?

**Comment by alextran1502** [comment]
We don't have a planned release date 😁

**Comment by auanasgheps** [comment]
@nick008a you opened a duplicate of this. Please close yours.

**Comment by Srylax** [comment]
> @nick008a you opened a duplicate of this FR. Please close yours.

I don't think it's a duplicate. Sure, both are used to delete local copies, but @nick008a FR only deletes a specific copy. The argument to have the server stable isn't as strong because it's not as destructive. To block @nick008a feature because of a non-stable server is just an annoyance because you can already delete photo via the native photo app.

**Comment by spupuz** [comment]
> crazy such a crucial feature is not in? bump

yeah :(

**Comment by 4NDREWV** [comment]
> crazy such a crucial feature is not in? bump

Please read the thread, folks

https://github.com/immich-app/immich/issues/165#issuecomment-1575164317

**Comment by LakhveerChahal** [comment]
Is there any alternative to this for the time being? I want to delete backed up photos on daily basis from my phone.

**Comment by kiwijunglist** [comment]
I would like to
- delete and down-sample images that have already been uploaded to server
- have a down-sampled local copy of images on server that are not on my phone.

**Comment by vinicentus** [comment]
The PR #4505 seems like it will be merged soon, and it would help mitigate this issue while we wait for the full solution.

**Comment by spupuz** [comment]
> The PR #4505 seems like it will be merged soon, and it would help mitigate this issue while we wait for the full solution.

yeah but this is a different think...

**Comment by vinicentus** [comment]
> yeah but this is a different think...

It seems to implement the manual deletion of already backed up photos, but not any automatic deletion.

**Comment by spupuz** [comment]
additionally it seems that is not yet merged correct?

**Comment by shaqman** [comment]
> > yeah but this is a different think...
> 
> It seems to implement the manual deletion of already backed up photos, but not any automatic deletion.

It's a good first step. Iterative building is key.

**Comment by spupuz** [comment]
> > > yeah but this is a different think...
> > 
> > 
> > It seems to implement the manual deletion of already backed up photos, but not any automatic deletion.
> 
> It's a good first step. Iterative building is key.

Yeah I imagine a pollssible option to remove automatically local copies of all backed up photo older than a week for example

**Comment by eUnisk** [comment]
New to immich here. I do want this feature to make immich our perfect local cloud.

**Comment by mchiappinam** [comment]
A possible solution to this issue would be to add an option for "select all," as we already have the option to delete only backed-up photos implemented.

**Comment by chrismin13** [comment]
Please, if this feature is implemented, consider that additional information **Portrait and Edited photos on iOS might be lost**. Google Photos does not delete Portrait photos, since when they're uploaded, they lose their adjustable focus and blur. Unfortunately, they don't detect edited photos, so if you clear the edited photos after uploading them, there is no way to revert them or change the edits.

I think it's critical that these are either fully backed up or not deleted by such a feature.

Other than that, very excited to see how this project involves, this was much needed. Thank you for all your hard work!

**Comment by alextran1502** [comment]
@CurlyFlow 

> Why should the focus be backup? 

That is the initial reason why the project is created in the first place, you can read more here https://immich.app/docs/overview/introduction

**Comment by mchiappinam** [comment]
Omg please guys lol

**Comment by v3DJG6GL** [comment]
> I dont see "Android backup solution"

I would recommend that you actually read that page....

**Comment by mchiappinam** [comment]
As a immigrant, I don't think you compreende english man. I respect your opinion but i disagree 100%. This project is amazing for photos management, it's like Google photos but self hosted, i have almost 1tb in photos/videos, and a bunch of family members also use Immich, and I can see/search photos as they were in my gallery. It's a free open source project because Alex is amazing, as well as every dev that contribute to this project. Now, if you're worried about how much money Alex should be making, maybe you're a financial advisor and didn't told us 😂, we might be missing that piece.
I suggest you to create a topic with your pricing opinion, or maybe create from scratch your photo management system and sell it. This topic is about removing photos on mobile that are already backed up.

**Comment by spupuz** [comment]
> As a immigrant, I don't think you compreende english man. I respect your opinion but i disagree 100%. This project is amazing for photos management, it's like Google photos but self hosted, i have almost 1tb in photos/videos, and a bunch of family members also use Immich, and I can see/search photos as they were in my gallery. It's a free open source project because Alex is amazing, as well as every dev that contribute to this project. Now, if you're worried about how much money Alex should be making, maybe you're a financial advisor and didn't told us 😂, we might be missing that piece. I suggest you to create a topic with your pricing opinion, or maybe create from scratch your photo management system and sell it. This topic is about removing photos on mobile that are already backed up.

yeah i opened it just to have an option to remove pics arleady backup up photos to immich.

**Comment by bo0tzz** [comment]
This thread has gotten out of hand, let's drop the discussion and keep it on topic folks.

**Comment by eUnisk** [comment]
> This thread has gotten out of hand, let's drop the discussion and keep it on topic folks.

Agree. 

IMO this project has done a perfect job in terms of backup photos. 

I hope the mobile app working with the server end could have a feature to free up space by safely deleting photos that are already backed up.  Here are my reasons:

- Mobile phones are the most popular photo-shooting devices nowadays.
- Many people use cloud services for the main purpose of extending the storage capacity.

**Comment by cristianku** [comment]
For the moment I am exporting manually the photos and videos from my phone and the manually uploading to my server to a folder where I can launch the CLI command to import them . This way I can flush the phone storage and keep it clean . On the other hand immich doesn't have the edit so it's far away from the apple icloud solution but i think it would be really nice to have such seamless experience like with icloud from apple. For the time being there are for me always two library's the iOS photo and the immich server seen as two separate entities and if I need to edit I download the photo to my phone edit it and somewhere later upload it again manually to the server. Not the best experience . Worth noticing when the possibility to revert to the original goes lost as someone mentioned

**Comment by DiegoOrtiz-npc** [comment]
The truth is that if it had this automatic deletion option, the application would gain much more.
Because I think the alternatives that currently exist are worse than this one.
This is more successful at all levels (web, mobile app, facial recognition)

**Comment by babaz8** [comment]
Can't wait to see this feature implemented

**Comment by spupuz** [comment]
My parents are old and have their phone full of images a feature like this as i requeested would be awesome

**Comment by bo0tzz** [comment]
@spupuz you keep repeatedly requesting this, but all that will serve to do is annoy people, it won't speed anything up. As has been stated a few times now, this feature will only be implemented once we approach a stable release.

**Comment by spupuz** [comment]
> @spupuz you keep repeatedly requesting this, but all that will serve to do is annoy people, it won't speed anything up. As has been stated a few times now, this feature will only be implemented once we approach a stable release.

I'm Sorry you are bothered, but this is just your opinion, i see many person  that like me has requested this functionality, additionally seems that immich is widely utilized and more than stable.

**Comment by lukebouch** [comment]
This feature is very much needed. Right now I'm having to go in to the files app on my GrapheneOS phone and delete all of the files manually. When I open the Immich app back up after deleting the files, it can't find them to preview them and I have to sign out of the app and sign back in. I have to do this every couple of weeks because storage is at a premium on my 128GB phone.

**Comment by shawzymoto** [comment]
Can i get an amen??

**Comment by jtagcat** [comment]
> Automatic delete of local assets based on some settings, some ideas below: 

Unsolicited input: Last accessed/viewed, for single media items, but possibly also recently changed albums.

Using interface pair sounds reasonable: One side defines a quota for clean-up (derived from used storage, free space on device, etc). With manual clean-up, quota should be larger (fewer files kept). The second is responsible for progressively deciding what to delete, by assigning weighed points based on all attributes. An alternative would be step-by-step interfaces (each being more 'brutal' in sacrifice conditions).


### GitHub Etiquette

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by proskater123789** [comment]
Would love to see this!

**Comment by bo0tzz** [comment]
There's nothing new to add to this request, so I'm going to lock the thread. If there are any updates they will be posted here.
